lib/nerves_hub_link/client.ex

```diff
@@ -118,139 +122,116 @@ defmodule NervesHubLink.Client do
   @callback identify() :: :ok
 
   @doc """
-  Optional callback to reboot the device when a firmware update completes
-
-  The default behavior is to call `Nerves.Runtime.reboot/0` after a successful update. This
-  is useful for testing and for doing additional work like notifying users in a UI that a reboot
-  will happen soon. It is critical that a reboot does happen.
+  Callback to reboot the device when a firmware update completes
   """
   @callback reboot() :: no_return()
 
-  @optional_callbacks [reconnect_backoff: 0, reboot: 0]
+  defmacro __using__(_opts) do
+    quote do
+      @behaviour NervesHubLink.Client
+      require Logger
 
-  @doc """
-  This function is called internally by NervesHubLink to notify clients.
-  """
-  @spec update_available(update_data()) :: update_response()
-  def update_available(data) do
-    case apply_wrap(mod(), :update_available, [data]) do
-      :apply ->
-        :apply
+      @impl true
```

Comment:
```suggestion
      @impl NervesHubLink.Client
```
By convention, I like to be explicit in the impl for more clarity and extra compiler help

Reply:
I'll change these back

Reply:
updated